File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/CubeInTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class CubeInTransformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		view.setPivotX(position > 0 ? 0 : view.getWidth());
		view.setPivotY(0);
		view.setRotationY(-90F * position);
	}

	@Override
	public boolean isPagingEnabled() {
		return true;
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/CubeInTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View
import me.shkschneider.skeleton.extensions.then

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class CubeInTransformer : ABaseTransformer() {

    override fun isPagingEnabled(): Boolean {
        return true
    }

    override fun onTransform(page: View, position: Float) {
        page.pivotX = ((position > 0) then 0 ?: page.width).toFloat()
        page.pivotY = 1.toFloat()
        page.rotationY = (-1).toFloat() * position
    }

}
